courses/machine_learning/deepdive2/structured/solutions/1_explore_babyweight.ipynb

```diff
@@ -0,0 +1,591 @@
+{
```

Comment:
<ul><li>Do we need to create the BigQuery client each time we issue a query? If not, just create one client at the beginning of the notebook and reuse it throughout the notebook.</li><li>Why are we calling these columns "valid"? From the query, it seems we are extracting all the columns from that table. If no real reason to call these columns "valid", let's rename the variables as follows: valid_columns_df -&gt; columns_df and valid_columns_set -&gt; columns_set</li></ul>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/training-data-analyst/pull/665/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

Reply:
<ul><li>Good suggestion, changed to create client just once.</li><li>It is the set of all valid columns that a user could try as part of the SQL Injection hacking attack prevention. I have valid in the name to reinforce that these are the only valid columns a user can choose from to make clear the SQL Injection prevention.</li></ul>